include/z64bgcheck.h

```diff
@@ -73,18 +71,39 @@ typedef struct {
     /* 0x10 */ s16 unk_10; // unused
 } BgCamFuncData; // size = 0x12
 
+// Macros for `WaterBox.properties`
+
+#define WATERBOX_BGCAM_INDEX_SHIFT 0
+#define WATERBOX_BGCAM_INDEX_MASK 0x000000FF
+#define WATERBOX_BGCAM_INDEX(properties) \
+    (((properties) >> WATERBOX_BGCAM_INDEX_SHIFT) & (WATERBOX_BGCAM_INDEX_MASK >> WATERBOX_BGCAM_INDEX_SHIFT))
+
+#define WATERBOX_LIGHT_INDEX_SHIFT 8
+#define WATERBOX_LIGHT_INDEX_MASK 0x00001F00
+#define WATERBOX_LIGHT_INDEX(properties) \
+    (((properties) >> WATERBOX_LIGHT_INDEX_SHIFT) & (WATERBOX_LIGHT_INDEX_MASK >> WATERBOX_LIGHT_INDEX_SHIFT))
+#define WATERBOX_LIGHT_INDEX_NONE 0x1F // warns and defaults to 0
+
+#define WATERBOX_ROOM_INDEX_SHIFT 13
+#define WATERBOX_ROOM_INDEX_MASK 0x0007E000
+#define WATERBOX_ROOM_INDEX(properties) \
+    (((properties) >> WATERBOX_ROOM_INDEX_SHIFT) & (WATERBOX_ROOM_INDEX_MASK >> WATERBOX_ROOM_INDEX_SHIFT))
+#define WATERBOX_ROOM_INDEX_ALL 0x3F // value for "room index" indicating "all rooms"
+
+#define WATERBOX_FLAG_19 (1 << 19)
+
+#define WATERBOX_PROPERTIES(bgCamIndex, lightIndex, roomIndex, setFlag19)       \
+    ((((bgCamIndex)&WATERBOX_BGCAM_INDEX_MASK) >> WATERBOX_BGCAM_INDEX_SHIFT) | \
+     (((lightIndex)&WATERBOX_LIGHT_INDEX_MASK) >> WATERBOX_LIGHT_INDEX_SHIFT) | \
+     (((lightIndex)&WATERBOX_ROOM_INDEX_MASK) >> WATERBOX_ROOM_INDEX_SHIFT) | ((setFlag19) ? WATERBOX_FLAG_19 : 0))
```

Comment:
the ternary for that last one is odd to me. shouldnt it just be the value? if you pass 0 then it will be 0

Reply:
I'm envisioning `true`/`false` to be passed as `setFlag19`, that seems more in line with other enums, but I can change the last arg to just be `flags` (and it would only be used for the single flag that exists)

Reply:
even if it was true or false, I dont see the purpose of the ternary. true or false will just be 0 or 1 anyway right? 
this might be some standard thing ive just never seen before, just seems weird to do extra stuff that isnt needed

Reply:
For example `SCENE_CMD_ROOM_BEHAVIOR` takes true/false for `showInvisActors` and `disableWarpSongs`, and sets (or doesn't set) the corresponding bit?
That's what `((setFlag19) ? WATERBOX_FLAG_19 : 0)` does,
or do you prefer `(((setFlag19) & 1) << WATERBOX_FLAG_19_SHIFT)` or something?

Reply:
yea I would just expect it to be set like everything else. at least ive never seen a boolean thats bitpacked like this be set with a ternary.
if its something we want to start doing then we can see how others feel about it, but afaik its not something we've ever done

Reply:
also should this be shifting left? this is packing the values in right

Reply:
Yeah it should, duh me

Reply:
id also add a space between the & on both sides

Reply:
The formatter made me (not) do it!
(I try to still format parts of .h files to help consistency)